lib/src/test/java/vmath/Mat3Test.java

```diff
@@ -20,6 +21,63 @@ void givenNoArgs_whenMat3Created_thenMatrixIsIdentity() {
         assertEquals(1.0f, m1.get(2, 2));
     }
 
+    @Test
+    void givenArrayArg_whenMat3Created_thenMatrixHasGivenValues() {
+        var arr = new float[] {
+            1.0f, 2.0f, 3.0f,
+            4.0f, 5.0f, 6.0f,
+            7.0f, 8.0f, 9.0f
+        };
+        var m = new Mat3(arr);
+
+        assertEquals(m.get(0, 0), 1.0f);
+        assertEquals(m.get(0, 1), 2.0f);
+        assertEquals(m.get(0, 2), 3.0f);
+
+        assertEquals(m.get(1, 0), 4.0f);
+        assertEquals(m.get(1, 1), 5.0f);
+        assertEquals(m.get(1, 2), 6.0f);
+
+        assertEquals(m.get(2, 0), 7.0f);
+        assertEquals(m.get(2, 1), 8.0f);
+        assertEquals(m.get(2, 2), 9.0f);
+    }
+
+    @Test
+    void givenArrayOfInvalidLength_whenMat3Created_thenThrowsException() {
+        var arr = new float[] {
+            1.0f, 2.0f, 3.0f,
+            4.0f, 5.0f, 6.0f,
+            7.0f, 8.0f, 9.0f, 0.0f
+        };
+
+        assertThrows(IllegalArgumentException.class, () -> new Mat3(arr));
+    }
+
+    @Test
+    void givenNaNValues_whenMat3Created_thenThrowsException() {
+        float[] arrOfNaNValues = {
+            Float.NaN, Float.NaN,
+            Float.NaN, Float.NaN
+        };
+
+        assertThrows(IllegalArgumentException.class, () -> new Mat3(arrOfNaNValues));
+
+        float[] arrWithNaN = {
+            1.0f, 2.0f,
+            Float.NaN, 3.0f
+        };
```

Comment:
This test should be checking for an ArithmeticException and the arrays must have 9 values, not 4.